cmd/gitops/add/profiles/cmd.go

```diff
@@ -0,0 +1,132 @@
+package profiles
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/go-resty/resty/v2"
+	"github.com/spf13/cobra"
+	"github.com/weaveworks/weave-gitops/cmd/internal"
+	"github.com/weaveworks/weave-gitops/pkg/flux"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"github.com/weaveworks/weave-gitops/pkg/models"
+	"github.com/weaveworks/weave-gitops/pkg/osys"
+	"github.com/weaveworks/weave-gitops/pkg/runner"
+	"github.com/weaveworks/weave-gitops/pkg/server"
+	"github.com/weaveworks/weave-gitops/pkg/services"
+	"github.com/weaveworks/weave-gitops/pkg/services/auth"
+	"github.com/weaveworks/weave-gitops/pkg/services/profiles"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+var opts profiles.AddOptions
+
+// AddCommand provides support for adding a profile to a cluster.
+func AddCommand(client *resty.Client) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "profile",
+		Aliases:       []string{"profiles"},
+		Short:         "Add a profile to a cluster",
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		Example: `
+		# Add a profile to a cluster
+		gitops add profile --name=podinfo --cluster=prod --version=1.0.0 --config-repo=ssh://git@github.com/owner/config-repo.git
+		`,
+		RunE: addProfileCmdRunE(client),
+	}
+
+	cmd.Flags().StringVar(&opts.Name, "name", "", "Name of the profile")
+	cmd.Flags().StringVar(&opts.Version, "version", "latest", "Version of the profile")
+	cmd.Flags().StringVar(&opts.ConfigRepo, "config-repo", "", "URL of external repository (if any) which will hold automation manifests")
+	cmd.Flags().StringVar(&opts.Cluster, "cluster", "", "Name of the cluster to add the profile to")
+	cmd.Flags().StringVar(&opts.Port, "port", server.DefaultPort, "Port the profiles API is running on")
```

Comment:
```suggestion
	cmd.Flags().StringVar(&opts.Port, "profiles-api-port", server.DefaultPort, "Port the profiles API is running on")
```

i wonder if its even possible to just have the code discover this, since its already querying k8s. perhaps an improvement for down the line